PubHub/PubHub.API/Controllers/UserController.cs

```diff
@@ -142,23 +143,91 @@ public async Task<IActionResult> DeleteUserAsync(int id)
                 .SingleOrDefaultAsync();
             if (accountId == default)
             {
-                return Problem(
+                return Results.Problem(
                     statusCode: StatusCodes.Status404NotFound,
                     detail: $"No user with ID: {id}");
             }
 
             // Delete account.
-            int updatedRows = await _context.Users
+            int updatedRows = await _context.Set<Account>()
                 .Where(u => u.Id == accountId)
                 .ExecuteUpdateAsync(u => u.SetProperty(u => u.IsDeleted, true));
             if (updatedRows > 0)
             {
-                return Ok();
+                return Results.Ok();
             }
 
-            return Problem(
+            // Unable to delete; report back.
+            Dictionary<string, object?> extensions = new()
+            {
+                ["UserId"] = id,
+                ["AccountId"] = accountId
+            };
+
+            return Results.Problem(
+                statusCode: StatusCodes.Status500InternalServerError,
+                detail: $"Unable to delete account (ID: {accountId}) of user (ID: {id}).",
+                extensions: extensions);
+        }
+
+        /// <summary>
+        /// Suspend the account of a specific user.
+        /// </summary>
+        /// <param name="id">ID of user.</param>
+        /// <response code="200">Success. The user was suspended.</response>
+        /// <response code="404">The user wasn't found.</response>
+        /// <response code="500">Unexpected error.</response>
+        [HttpDelete(Name = "SuspendUser")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<IResult> SuspendUserAsync(int id)
+        {
+            const string accountTypeName = "suspended";
+
+            // Get account type ID.
+            var accountTypeId = await _context.Set<AccountType>()
+                .FirstOrDefaultAsync(a => a.Name.Equals(accountTypeName, StringComparison.InvariantCultureIgnoreCase));
+            if (accountTypeId == default)
```

Comment:
Again `default` should not be used in this context. It's difficult to know what the default value is wihtout actually looking at the type.

I suggest cheching for `null` if that is the intend.

Reply:
I discovered that the `accountTypeId` wasn't actually the ID but the whole entity. It should be fixed to be only the ID now.

Reply:
> We cannot check for null as the type is `int`. Should we just check with `accountTypeId == 0` instead? I thought using `default` would make it more readable when reading it like so:
> 
> 1. `accountTypeId` is first or default.
> 2. If `accountTypeId` is default; do this.

Aah, I see. I thought the ID was an `int?`.
I'd actually rather have it be specific, but I don't know. It's just my opinion.
@jeongoks What's your take on this?

I could go both ways tbh. I'd just prefer it if we're explicit where we can be 🤔

Reply:
Shouldn't we just go with 0 then? I will be doing something similar in the PublisherController but with `-1` as a second result value. Using `0` instead of `default` there, will make it look more pretty 🙃

Reply:
I would prob. do something like `const int INVALID_ID = 0` as that's more readable.
And I would do so for -1 as well in relation to whatever you're using it for 🤔

Reply:
Yes, I agree with that solution. 🙂 I will define some local constants to make the code more readable.

Reply:
Seems like you guys came to a conclusion on your own.

However, I do also like being more specific when we're doing checks.